client-app/shared/wishlists/components/add-to-wishlists-dialog.vue

```diff
@@ -1,70 +1,135 @@
+          <div class="py-3 px-6 font-bold text-15 leading-5 bg-[color:var(--color-add-wishlist-dialog-subtitle-bg)]
+          sm:py-2.5">
+            {{ $t("shared.wishlists.add_to_wishlists_dialog.already_in_the_lists") }}
           </div>
 
-          <VcCheckbox v-else v-model="selectedListsIds" :value="list.id" :disabled="loading" class="px-6 py-4">
-            <span class="text-base line-clamp-1" :class="{ 'text-gray-500': !selectedListsIds.includes(list.id!) }">
-              {{ list.name }}
-            </span>
-          </VcCheckbox>
-        </li>
-      </ul>
+          <ul>
+            <li v-for="(list, index) in listsContain" :key="list.id" class="px-6 py-4 sm:pt-4 sm:pb-3 last:sm:pb-7">
+              <VcCheckbox
+                model-value
+                :value="list.id"
+                :disabled="loading"
+                @update:modelValue="listsRemoveUpdate(list.id || '', $event)"
+              >
+                <span class="text-base line-clamp-1 font-medium sm:text-15">
+                  {{ list.name }}
+                </span>
+              </VcCheckbox>
+            </li>
+          </ul>
+        </template>
+
+        <div class="flex justify-between py-3 px-6 bg-[color:var(--color-add-wishlist-dialog-subtitle-bg)] sm:py-2.5">
+          <div class="font-bold text-15">
+            {{ $t("shared.wishlists.add_to_wishlists_dialog.add_to_other_lists") }}
+          </div>
+          <button
+            class="flex items-center text-sm font-bold cursor-pointer text-[color:var(--color-link)]"
+            :class="{ 'text-gray-400 cursor-not-allowed': creationButtonDisabled }"
+            @click="addInput"
+            :disabled="creationButtonDisabled"
+          >
+            <svg
+              class="mr-2 w-3.5 h-3.5 text-[color:var(--color-primary)]"
+              :class="{ 'text-gray-400': creationButtonDisabled }"
+            >
+              <use href="/static/images/plus.svg#main" />
+            </svg>
+            {{ $t("shared.wishlists.add_to_wishlists_dialog.add_new_list") }}
+          </button>
+        </div>
+
+        <transition-group name="list-input" tag="ul" class="pt-2 sm:pt-0.5">
+          <li
+            v-for="(input, index) in inputs"
+            :key="index"
+            class="flex items-start px-6 list-input-item first:pt-3 first:sm:pt-4"
+          >
+            <button class="relative pt-3" @click="removeInput(index)">
+              <VcCheckbox model-value class="relative" />
+              <div class="absolute inset-0"></div>
+            </button>
+            <VcInput
+              class="flex-grow ml-2.5 mr-3.5"
+              :class="{ 'mb-4.5': !input.errorMessage }"
+              v-model="input.listName"
+              :is-disabled="loading"
+              :maxlength="25"
+              is-required
+              :error-message="input.errorMessage"
+            ></VcInput>
+            <button class="pt-3.5" @click="removeInput(index)">
```

Comment:
With padding the button has a strange clickable area: the X mark itself and some empty space above. Maybe it's better to use margin instead?

Reply:
done